proposals/NNNN-Update-vehicle-data-TurnSignal.md

```diff
@@ -0,0 +1,135 @@
+
+```xml
+<struct name="TurnSignalData" since="X.x">
+	<param name="turnSignal" type="TurnSignal" mandatory="false" since="X.x">
+		<description>See TurnSignal</description>
+	</param>	
+	<param name="hazardLights" type="VehicleDataStatus" mandatory="false" since="X.x">
+		<description>Status of hazard lights</description>
+	</param>
+</struct>
+```
+
+#### Update the following parameter to these function requests:
+* `SubscribeVehicleData`
+* `UnsubscribeVehicleData`
+* `GetVehicleData`
+
+```xml
+<param name="turnSignal" type="Boolean" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+#### Update the following parameter to these function responses:
+* `SubscribeVehicleData`
+* `UnsubscribeVehicleData`
+
+```xml
+<param name="turnSignal" type="VehicleDataResult" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+#### Update the following parameter to these function responses:
+* `GetVehicleData`
+* `OnVehicleData`
+
+```xml
+<param name="turnSignal" type="TurnSignalData" mandatory="false" since="X.x">
+	<description>See TurnSignalData</description>
+	<history>
+		<param name="turnSignal" type="TurnSignal" mandatory="false" since="5.0" until="X.x">
+			<description>See TurnSignal</description>
+		</param>
+	</history>
+</param>
+```
+
+### Updates in HMI_API:
+
+#### Add new struct enum `TurnSignalData` in `Common` interface: 
+
+```xml
+<struct name="TurnSignalData">
+	<param name="turnSignal" type="Common.TurnSignal" mandatory="false">
+		<description>See TurnSignal</description>
+	</param>	
+	<param name="hazardLights" type="Common.VehicleDataStatus" mandatory="false">
+		<description>Status of hazard lights</description>
+	</param>
+</struct>
+
+```
+
+#### Update the following parameter to these function requests:
+* `SubscribeVehicleData`
+* `UnsubscribeVehicleData`
+* `GetVehicleData`
+
+```xml
+<param name="turnSignal" type="Boolean" mandatory="false">
+	<description>See Common.TurnSignalData</description>	
+</param>
+```
+
+#### Update the following parameter to these function responses:
+* `SubscribeVehicleData`
+* `UnsubscribeVehicleData`
+
+```xml
+<param name="turnSignal" type="Common.VehicleDataResult" mandatory="false">
+	<description>See Common.TurnSignalData</description>	
+</param>
+```
+
+#### Update the following parameter to these function responses:
+* `GetVehicleData`
+* `OnVehicleData`
+
+```xml
+<param name="turnSignal" type="Common.TurnSignalData" mandatory="false">
+	<description>See TurnSignalData</description>
+	</history>
+</param>
+```
+
+## Potential downsides
+
+Since this is a breaking change, HMI would need to be updated as well.
+ 
+## Impact on existing code
+
+* SDL Core needs to be updated as per new API.
+* HMI needs to be updated to support refactored structures.
```

Comment:
Please provide iOS and Java Suite code impact, as these are listed as impacted platforms.

Reply:
Done.